### PR TITLE
add support for helm catalogs

### PR DESCRIPTION
**Pull request by wlan0**
@joshwget

Doing things in 1 transaction takes a long long time...

### CODE REVIEW

File: manager/traverse.go [hunk 1]
```diff
@@ -9,12 +9,73 @@ import (
...
 	"gopkg.in/yaml.v2"
 )
 
-func traverseFiles(repoPath string) ([]model.Template, []error, error) {
+func traverseFiles(repoPath, kind string) ([]model.Template, []error, error) {
+	if kind == "" || kind == "rancher" {
+		return traverseGitFiles(repoPath)
+	}
+	if kind == "k8s" {
+		index, err := helm.LoadIndex(repoPath)
```
Comment (joshwget) [inline manager/traverse.go]: Can you move the stuff under `if kind == "k8s" {` into a separate function like it is for `rancher`?

File: repo.json [hunk 2]
```diff
@@ -1,10 +1,15 @@
 {
   "catalogs": {
     "library": {
-      "url": "https://github.com/rancher/rancher-catalog"
+      "url": "https://github.com/rancher/rancher-catalog",
+      "kind": "rancher"
```
Comment (joshwget) [inline repo.json]: Can you put this on community catalog too?

File: repo.json [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
     },
     "community": {
       "url": "https://github.com/rancher/community-catalog"
+    },
+    "helm": {
```
Comment (joshwget) [inline repo.json]: Can you move the Helm stuff to a separate `helm.json`?

**Comment by joshwget** [comment]
LGTM other than a few small things

**Comment by wlan0** [comment]
@joshwget Updated code according to your comments

**Comment by joshwget** [comment]
Okay, real LGTM this time. I'll check to see if @vincent99 or @ibuildthecloud want to review.

**Comment by joshwget** [comment]
Can you setup a test catalog and add integration tests? (Similar to https://github.com/rancher/test-catalog)

**Pull request merged**